addons/point_of_sale/models/pos_config.py

```diff
@@ -463,9 +463,10 @@ def write(self, vals):
         opened_session = self.mapped('session_ids').filtered(lambda s: s.state != 'closed')
         if opened_session:
             forbidden_fields = []
-            for key in self._get_forbidden_change_fields():
-                if key in vals.keys():
-                    field_name = self._fields[key].get_description(self.env)["string"]
+            field_names = self.fields_get(allfields=self._get_forbidden_change_fields(), attributes=["string"])
+            for fname, info in field_names.items():
```

Comment:
If `vals` is correct, this retrieves field descriptions for nothing.
Wouldn't it be better to avoid `fields_get()` when there is no problem?
I suggest something like:
```py
forbidden_fields = [fname for fname in self._get_forbidden_change_fields() if fname in vals]
if forbidden:
    field_infos = self.fields_get(forbidden_fields, attributes=['string'])
    raise UserError(_(
        ...,
        ", ".join(info['string'] for info in field_infos.values())
    )
```